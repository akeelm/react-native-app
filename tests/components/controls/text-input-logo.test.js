import React from 'react';
import TextInputLogo from '../../../app/components/controls/text-input-logo';

import {
	_validateEmail,
	_minimumPasswordLength
} from '../../../app/helpers/validation';

import Icon from 'react-native-vector-icons/FontAwesome';
import renderer from 'react-test-renderer';


it('should render', () => {
    const component = renderer.create(
        <TextInputLogo title='E-mail' logo='user-o'
            onChangeMethod={jest.fn()}
            validate={_validateEmail}
        />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

it('focus should change styling', () => {
    const component = renderer.create(
        <TextInputLogo title='E-mail' logo='user-o'
            onChangeMethod={jest.fn()}
            validate={_validateEmail}
        />
    );
    component.getInstance().toggleFocus();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should display an error message', () => {
    const component = renderer.create(
        <TextInputLogo title='E-mail' logo='user-o'
            onChangeMethod={jest.fn()}
            validate={_validateEmail}
        />
    )
    component.getInstance().setState({ error: true, errorMessage: 'An error message' });
    expect(component.toJSON()).toMatchSnapshot();
  });