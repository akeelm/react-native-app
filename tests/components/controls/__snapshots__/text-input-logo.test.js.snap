// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`focus should change styling 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#5fba7d",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#5fba7d",
            "fontSize": 20,
          },
          Object {
            "padding": 10,
          },
          Object {
            "fontFamily": "awesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="E-mail"
      returnKeyType="next"
      secureTextEntry={undefined}
      style={
        Object {
          "borderBottomWidth": 0,
          "flex": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`should display an error message 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#ccc",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 20,
          },
          Object {
            "padding": 10,
          },
          Object {
            "fontFamily": "awesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="E-mail"
      returnKeyType="next"
      secureTextEntry={undefined}
      style={
        Object {
          "borderBottomWidth": 0,
          "flex": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#f38383",
      }
    }
  >
    An error message
  </Text>
</View>
`;

exports[`should render 1`] = `
<View
  style={
    Object {
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#ccc",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <TextInput
      allowFontScaling={true}
      editable={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="E-mail"
      returnKeyType="next"
      secureTextEntry={undefined}
      style={
        Object {
          "borderBottomWidth": 0,
          "flex": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
